tests/init/neg/local-warm4.scala

```diff
@@ -0,0 +1,31 @@
+object localWarm {
+    abstract class Foo(x: Int) {
+        def increment(): Unit
+    }
+
+    class A(x: Int) extends Foo(x) {
+        var y = x
+        override def increment(): Unit = y = y + 1
+        increment()
+        val b = new B(y)
+    }
+
+    class B(x: Int) extends A(x) {
+        var a: A = this
+        override def increment(): Unit = {
+            def updateA(): Unit = {
+                val newA = new A(y)
+                a = newA // error
+            }
+            y = y + 1
+            updateA()
+        }
+        if y < 10 then increment()
+        val z = b.y
+    }
+    val a = new A(5)
+}
+
+
+
+
```

Comment:
Minor: useless empty lines.

Reply:
I would suggest inventing more tests that demonstrate the capture of local non-hot definitions in functions and local classes.